### PR TITLE
Add traits support

### PR DESCRIPTION
**Pull request by v-kolesnikov**
Issue: #10 

Hi team!
This is a draft implementation of traits supports. There are a lot of places for optimisation and refactoring here, but it works in general. I'll be glad to hear any comments and tips.

TODO:
- [ ] Add documentation to rom-rb.org site

### CODE REVIEW

File: spec/integration/rom/factory_spec.rb [hunk 1]
```diff
@@ -243,10 +243,114 @@
...
+      end
+
+      jane = factories.structs[:user, :jane_doe]
+
+      expect(jane.first_name).to eql('Jane')
+      expect(jane.last_name).to eql('Doe')
+      expect(jane.email).to eql('jane@doe.org')
+    end
+
+    it 'allows to define traits for persisted' do
```
Comment (cutalion) [inline spec/integration/rom/factory_spec.rb]: what is this test for?

Comment (v-kolesnikov) [reply spec/integration/rom/factory_spec.rb]: [Persistable](https://github.com/rom-rb/rom-factory/pull/18/files#diff-8b6c33d5b26232d866a729facfb35a68) - is a different builder.

Comment (cutalion) [reply spec/integration/rom/factory_spec.rb]: Oh, I just didn't notice the difference between this test and the few above.
Now I see - there are `factories.struct[]` and `factories[]` calls

**Comment by solnic** [comment]
Great job man, thank you!

**Pull request merged**